### ISSUE TITLE
Commandline which does not fit screen is not redrawn correctly

### ISSUE THREAD

**Issue by krobelus** [fish-shell/fish-shell#7296]
Steps to reproduce:

1. Create a comandline that exceeds the terminal height:

```fish
bind \cg 'commandline -i "echo \'" (seq 1 $LINES)/$LINES "\'"'
```

2. Hit <kbd>Control</kbd>+<kbd>G</kbd>

3. Observe that fish fails to draw the last line

4. Type something. The cursor moves but no characters are echoed.

Scrolling up to the prompt I see that fish has started drawing the commandline at the wrong location, resulting in this screen.

```fish
$ echo '
  1/59


  2/59
  3/59
```

Happens at least on 3.1.2 and current master.

**Comment by laserscout** [comment]
I would be really interested in at least a temporary solution for this. Is it so much intertwined to the shell's core making the issue inevitable?

**Comment by esn89** [comment]
For something that's opened for 4 years, is there a fix?  A workaround?  Any type of mention would be nice.  Even if no known fix

**Comment by krobelus** [comment]
as mentioned on [reddit](https://old.reddit.com/r/fishshell/comments/1g86iof/fish_shell_pasting_text_blank_screen/), workarounds are `alt-e` (open command in external editor) or `ctrl-x` (copy command to clipboard)

**Comment by chapmanjacobd** [comment]
might be somewhat related, maybe this can inspire an idea for how to fix: https://gist.github.com/EgZvor/0a6a1f4252e74f322073b07a5f15af94

**Comment by krobelus** [comment]
it's already fixed but yeah opening an editor with the (last) command output is a nice feature (available in terminals that support OSC 133). The most obvious UI for that is to use the clipboard to copy the command over. But I guess we can try combining `alt-e` with that somehow, which also reduces the number of shortcuts to remember

**Comment by amab8901** [comment]
I'm still getting this issue, idk why you closed it

**Comment by esn89** [comment]
This is fix is coming in the rust rewrite of Fish 4.0.0 right?

**Comment by chapmanjacobd** [comment]
Yeah it's already in Fish 4.0 beta 1:

- Notes: https://github.com/fish-shell/fish-shell/releases/tag/4.0b1
- Ubuntu: https://launchpad.net/~fish-shell/+archive/ubuntu/beta-4
- Fedora: https://software.opensuse.org/download/package?package=fish&project=shells%3Afish%3Abeta%3A4
- Other: https://universal-blue.discourse.group/t/how-do-i-install-the-beta-version-of-fish-shell-rust-version/6255/3
